src/plan_bot.py

```diff
@@ -372,13 +385,11 @@ def process_flags(text):
     Identifies flags in the text. Removes the flags from the text and
     returns the tuple (remaining_text, options).
     """
-    options = {"parent": False}
-
-    match = re.match(
-        r"^(?:--parent|--tell-parent)[^-\w]+(.*)$", text, re.IGNORECASE | re.MULTILINE
-    )
-    if match:
-        options["parent"] = True
-        text = match.group(1)
-
-    return (text, options)
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument("--parent", "--tell-parent", action="store_true")
+    parser.add_argument("--why-warren", action="store_true")
+    parser.add_argument("rest", nargs=argparse.REMAINDER)
+    options, unknown = parser.parse_known_args(text.split())
+    remaining_text = " ".join(options.rest)
+    del options.rest  # we don't need this, and removing it makes testing easier
+    return (remaining_text, vars(options))
```

Comment:
Oh, I've never seen `vars` used before! Neat. Out of curiosity, is there an advantage versus `dict(options)`?

Reply:
They do different things.   `dict` takes a list of keyword arguments and turns them into a dictionary.   `vars` just calls the `__dict__` method of an object and returns its value.  It's an abstract way of letting an object determine its own dict representation.  In this case, `argparse.Namespace` gives you a dictionary of your options.

Reply:
☄